source/packages/services/events-processor/src/filter/filter.service.ts

```diff
@@ -190,7 +190,7 @@ export class FilterService {
     private async listSubscriptionsForEvent(
         ev: CommonEvent,
         subscriptionMap: { [key: string]: SubscriptionItem[] }
-    ) {
+    ): Promise<SubscriptionItem[]> {
```

Comment:
This should specify that it can return an undefined value. I also wonder, is there any value to returning an undefined value from this function? Why not return undefined as an empty array?

Reply:
I updated it. Also, you are correct about the undefined value. I did not want to correct that here and the multiple other places it happens in this CR. This function, it's callers, and the functions it calls should be refactored in a few ways. It's in my notes of things to be addressed.